L1Trigger/L1TCaloLayer1/src/UCTCard.hh

```diff
@@ -0,0 +1,68 @@
+#ifndef UCTCard_hh
+#define UCTCard_hh
+
+#include <vector>
+
+#include "UCTGeometry.hh"
+
+class UCTRegion;
+
+class UCTCard {
+public:
+
+  UCTCard(uint32_t crt, uint32_t crd);
+
+  virtual ~UCTCard();
+
+  // To set up event data before processing
+
+  const std::vector<UCTRegion*>& getRegions() const {return regions;}
+  const UCTRegion* getRegion(uint32_t rgn) const {return regions[rgn];}
+  const UCTRegion* getRegion(UCTRegionIndex r) const;
+
+  // To process event
+
+  bool clearEvent();
+  bool setECALData(UCTTowerIndex t, bool ecalFG, uint32_t ecalET);
+  bool setHCALData(UCTTowerIndex t, uint32_t hcalFB, uint32_t hcalET);
+  bool process();
+
+  // More access functions
+
+  const uint32_t getCrate() const {return crate;}
+  const uint32_t getCard() const {return card;}
+
+  const uint32_t et() const {return cardSummary;}
+
+  friend std::ostream& operator<<(std::ostream&, const UCTCard&);
+
+private:
+
+  // No default constructor is needed
+
+  UCTCard();
+
+  // No copy constructor is needed
+
+  UCTCard(const UCTCard&);
+
+  // No equality operator is needed
+
+  const UCTCard& operator=(const UCTCard&);
+
+  // Helper functions
+
+  const UCTRegion* getRegion(bool negativeEta, uint32_t caloEta, uint32_t caloPhi) const;
+
+  // Owned card level data 
+
+  uint32_t crate;
+  uint32_t card;
+
+  std::vector<UCTRegion*> regions;
```

Comment:
> We don't want to copy regions - we need to set the data of existing UCTRegion objects and similarly for all other objects created to represent the hardware. I am sure there are other ways of accomplishing this than through pointers, but this is what I chose. I would imagine that making this change will take longer to debug, so I suggest postponing to a later update.

The easiest solution would be to hold the regions in a vector by value
```c++
std::vector<UCTRegion> regions;
```
which you can fill in a similar manner in the constructor
```c++
    // Negative eta side
    regions.emplace_back(crate, card, true, rgn);
    // Positive eta side
    regions.emplace_back(crate, card, false, rgn);
```

Personally, I don't think this is critical request, and for the moment the code could go in as it is, in this regard.

I would still fix the namespace and #define vs. constexpr issue, though.